tests/test_run_haplotag.py

```diff
@@ -111,6 +111,578 @@ def test_haplotag_cli_parser(tmp_path):
+chr1_PS2_H2         -- supplem, cigar: 13868S1424M4D655M11500S, flag = 2048
+chr1_rcPS1_sub_H2   -- supplem, cigar: 8452S2673M2D375M15947S, flag = 2064
+chr2_PS1_H1         -- supplem, cigar: 18995S5407M3045S, flag = 2048
+chr1_rPS1_sub_H1    -- supplem, cigar: 2673M4D373M24401S, flag = 2064
+"""
+
+def test_run_haplotag_supplementary_skip(tmp_path):
+    var_file = "tests/data/supplementary_strategy_test.grch38.vcf.gz"
+    alignment_file = "tests/data/supplementary_strategy_test.grch38.bam"
+
+    out_bam_default_strategy = tmp_path / "output.default_haplotag_strategy.bam"
+    out_bam_explicit_skip = tmp_path / "output.explicit_skip_strategy.bam"
+
+    run_haplotag(variant_file=var_file,
+                 alignment_file=alignment_file,
+                 output=out_bam_default_strategy,
+                 ignore_read_groups=True,
+                 )
+
+    run_haplotag(variant_file=var_file,
+                 alignment_file=alignment_file,
+                 output=out_bam_explicit_skip,
+                 ignore_read_groups=True,
+                 supplementary_strategy=SupplementaryHaplotaggingStrategy.SKIP)
+
+    for a1, a2 in zip(pysam.AlignmentFile(out_bam_default_strategy), pysam.AlignmentFile(out_bam_explicit_skip)):
+        assert a1.query_name == a2.query_name
+        if a1.is_supplementary:
+            assert not a1.has_tag("HP")
+            assert not a2.has_tag("HP")
+            assert not a1.has_tag("PS")
+            assert not a2.has_tag("PS")
+        if a1.query_name == "R1" and not a1.is_supplementary:
+            assert not a1.has_tag("HP")
+            assert not a2.has_tag("HP")
+            assert not a1.has_tag("PS")
+            assert not a2.has_tag("PS")
+        elif a1.query_name == "R2" and not a1.is_supplementary:
+            assert a1.has_tag("HP")
+            assert a2.has_tag("HP")
+            assert a1.get_tag("HP") == a2.get_tag("HP")
+            assert a1.get_tag("HP") == 1
+            assert a1.has_tag("PS")
+            assert a2.has_tag("PS")
+            assert a1.get_tag("PS") == a2.get_tag("PS")
+            assert a1.get_tag("PS") == 16849384
+
+
+def test_run_haplotag_supplementary_copy_primary_no_strand_match_permissive_distance(tmp_path):
+    var_file = "tests/data/supplementary_strategy_test.grch38.vcf.gz"
+    alignment_file = "tests/data/supplementary_strategy_test.grch38.bam"
+
+    out_bam_copy_primary_strategy = tmp_path / "output.copy_primary.bam"
+
+    run_haplotag(variant_file=var_file,
+                 alignment_file=alignment_file,
+                 output=out_bam_copy_primary_strategy,
+                 ignore_read_groups=True,
+                 supplementary_strategy=SupplementaryHaplotaggingStrategy.COPY_PRIMARY,
+                 supplementary_strand_match=False,
+                 supplementary_distance_threshold=1_000_000)
+
+    a: pysam.AlignedSegment
+    for a in pysam.AlignmentFile(out_bam_copy_primary_strategy):
```

Comment:
Please always ensure the file gets closed correctly by using a context manager `with pyam.AlignmentFile(...) as ...:`).

Reply:
fixed